src/main.rs

```diff
@@ -33,16 +52,34 @@ fn main() {
 fn main() {
     // env_logger::init().unwrap();
 
+    let matches = App::new("ic-http-lambda")
+        .args(&[Arg::new("force-canister-id")
+            .about("Sets the canister id to use instead of parsing from the url.")
+            .takes_value(true)
+            .short('c')
+            .long("force-canister-id")
+            .default_value("")])
```

Comment:
I don’t think you need to pass `default_value` here. If the flag is not used, then `value_of` below will return `None`; that’s a cleaner way to detect “no canister id”.

Reply:
Yup, I thought so too, but got entire days-worth of compiler errors when trying to pass around `None`, so I said screw it and made it so the value was always there.

Reply:
I’m also consider myself to be new to Rust. Let me give it a shot, maybe we can both learn.